src/olympia/abuse/tasks.py

```diff
@@ -0,0 +1,49 @@
+from datetime import datetime, timedelta
+
+from django.db.models import Count, F, OuterRef, Q, Subquery
+
+from olympia import amo
+from olympia.abuse.models import AbuseReport
+from olympia.addons.models import Addon
+from olympia.amo.celery import task
+from olympia.reviewers.models import NeedsHumanReview, UsageTier
+
+
+@task
+def flag_high_abuse_reports_addons_according_to_review_tier():
+    usage_tiers = UsageTier.objects.filter(
+        # Tiers with no upper adu threshold are special cases with their own
+        # way of flagging add-ons for review (either notable or promoted).
+        upper_adu_threshold__isnull=False,
+        # Need a abuse reports ratio threshold to be set for the tier.
+        abuse_reports_ratio_threshold_before_flagging__isnull=False,
+    )
+
+    tier_filters = Q()
+    for usage_tier in usage_tiers:
+        tier_filters |= Q(
+            average_daily_users__gte=usage_tier.lower_adu_threshold,
+            average_daily_users__lt=usage_tier.upper_adu_threshold,
+            abuse_reports_count__gte=F('average_daily_users')
+            * usage_tier.abuse_reports_ratio_threshold_before_flagging
+            / 100,
+        )
+    if not tier_filters:
+        return
+
+    abuse_reports_count_qs = (
+        AbuseReport.objects.values('guid')
+        .filter(guid=OuterRef('guid'), created__gte=datetime.now() - timedelta(days=14))
+        .annotate(cnt=Count('*'))
+        .values('cnt')
+        .order_by()
+    )
+    qs = (
+        Addon.unfiltered.exclude(status=amo.STATUS_DISABLED)
+        .filter(type=amo.ADDON_EXTENSION)
+        .annotate(abuse_reports_count=Subquery(abuse_reports_count_qs))
+        .filter(tier_filters)
+    )
```

Comment:
I initially did it with manual SQL... but it was kinda awful to read, so I gave the ORM a shot... Also almost refactored the whole thing to share more code with `flag_high_hotness_according_to_review_tier()` but decided against it because it would not help in the long run IMHO.